Source/Fuse.Common/Internal/Bitmap.uno

```diff
@@ -0,0 +1,501 @@
+						throw outerException;
+					}
+				}
+			}
+		}
+
+		public static CPlusPlusHelpers.NativeBitmapHandle LoadFromByteArray(string pathHint, byte[] bytes)
+		@{
+			uBase::Auto<uBase::BufferPtr> buffer = new uBase::BufferPtr(bytes->Ptr(), bytes->Length(), false);
+			uBase::Auto<uBase::BufferStream> stream = new uBase::BufferStream(buffer, true, false);
+			return @{CPlusPlusHelpers.LoadFromNativeStream(string, NativeStreamHandle):Call(pathHint, stream)};
+		@}
+
+		public static CPlusPlusHelpers.NativeBitmapHandle LoadFromXliStream(string pathHint, CppXliStream stream)
+		{
+			var nativeStream = extern<NativeStreamHandle>(stream)"$0->_handle";
+			return LoadFromNativeStream(pathHint, nativeStream);
+		}
+
+		static int GetWidth(NativeBitmapHandle nativeBitmap)
+		@{
+			return nativeBitmap->GetWidth();
+		@}
+
+		static int GetHeight(NativeBitmapHandle nativeBitmap)
+		@{
+			return nativeBitmap->GetHeight();
+		@}
+
+		public static int2 GetSize(NativeBitmapHandle nativeBitmap)
+		{
+			return new int2(GetWidth(nativeBitmap), GetHeight(nativeBitmap));
+		}
+
+		public static int ReadPixel(NativeBitmapHandle nativeBitmap, int x, int y)
+		@{
+			uBase::Vector4u8 color = nativeBitmap->GetPixelColor(x, y);
+			return color.Z | (color.Y << 8) | (color.X << 16) | (color.W << 24); // encode as 0xAARRGGBB
+		@}
+	}
+
+	[extern(CPLUSPLUS) Require("Header.Include", "uImage/Bitmap.h")]
+	public sealed class Bitmap : IDisposable
+	{
+		public void Dispose()
+		{
+			debug_log "dispose coming in..";
+			if defined(Android)
+				AndroidHelpers.Recycle(NativeBitmap);
+			else if defined(iOS)
+				IOSHelpers.ReleaseImage(NativeImage);
+			else
+				build_error; // TODO!
+		}
+
+		readonly int2 _size;
+		public int2 Size { get { return _size; } }
+
+		extern(Android) public readonly Java.Object NativeBitmap;
+		extern(Android) protected Bitmap(Java.Object nativeBitmap)
+		{
+			NativeBitmap = nativeBitmap;
+			_size = AndroidHelpers.GetSize(nativeBitmap);
+		}
+
+		extern(iOS) public readonly IntPtr NativeImage;
+		extern(iOS) protected Bitmap(IntPtr nativeImage)
+		{
+			NativeImage = nativeImage;
+			_size = IOSHelpers.GetSize(nativeImage);
+		}
+
+		extern(CIL) public readonly System.Drawing.Bitmap NativeBitmap;
+		extern(CIL) protected Bitmap(System.Drawing.Bitmap nativeBitmap)
+		{
+			NativeBitmap = nativeBitmap;
+			_size = new int2(nativeBitmap.Width, nativeBitmap.Height);
+		}
+
+		extern(!Android && !iOS && CPLUSPLUS) internal readonly CPlusPlusHelpers.NativeBitmapHandle NativeBitmap;
+		extern(!Android && !iOS && CPLUSPLUS) protected Bitmap(CPlusPlusHelpers.NativeBitmapHandle nativeBitmap)
+		{
+			NativeBitmap = nativeBitmap;
+			_size = CPlusPlusHelpers.GetSize(nativeBitmap);
+		}
+
+		static Bitmap LoadFromBundleFile(BundleFile bundleFile)
+		{
+			if (bundleFile.BundlePath.EndsWith(".gif"))
+			{
+				// TODO: Fuse.Diagnostic.InternalError
+				debug_log "Gif files are not fully supported!";
```

Comment:
debug_log is probably not what you want here.
Either ` Uno.Diagnostics.Debug.Log()`, or `Fuse.Diagnostics.UserWarning()`  is going to be more appropriate.